seacliff/tests/test_rubin_sky_wcs.py

```diff
@@ -0,0 +1,147 @@
+    cd_matrix = np.dot(cd_matrix, rotmat)
+    ra = rng.uniform(low=0, high=2.0 * np.pi)
+    dec = np.arcsin(rng.uniform(-1, 1))
+    crval = lsst.geom.SpherePoint(ra, dec, lsst.geom.radians)
+    return lsst.afw.geom.makeSkyWcs(
+        crpix=crpix,
+        crval=crval,
+        cdMatrix=cd_matrix,
+    )
+
+
+def _test_pos(wcs1, wcs2, shift=None):
+    if shift is not None:
+        dx = shift.x
+        dy = shift.y
+    else:
+        dx = 0
+        dy = 0
+    ra1, dec1 = wcs1.xyToradec(
+        np.array([10, 9]),
+        np.array([78, 55]),
+        units=galsim.degrees,
+    )
+    ra2, dec2 = wcs2.xyToradec(
+        np.array([10, 9]) + dx,
+        np.array([78, 55]) + dy,
+        units=galsim.degrees,
+    )
+    assert_allclose(ra1, ra2)
+    assert_allclose(dec1, dec2)
+
+    x1, y1 = wcs1.radecToxy(ra1, dec1, units=galsim.degrees)
+    x2, y2 = wcs2.radecToxy(ra2, dec2, units=galsim.degrees)
+    assert_allclose(x1, x2 - dx)
+    assert_allclose(y1, y2 - dy)
+
+    assert_allclose(x1, np.array([10, 9]))
+    assert_allclose(y1, np.array([78, 55]))
+
+
+def test_rubin_sky_wcs_correct():
+    x0 = 10
+    y0 = 2
+    x = np.array([10.0, 9.0])
+    y = np.array([78.0, 55.0])
+
+    wcs = seacliff.RubinSkyWCS(_make_wcs(10), origin=galsim.PositionD(x0, y0))
+    ra, dec = wcs.xyToradec(x, y, units=galsim.degrees)
+
+    ra_rubin, dec_rubin = wcs.wcs.pixelToSkyArray(x - x0 - 1, y - y0 - 1, degrees=True)
+    assert_allclose(ra, ra_rubin)
+    assert_allclose(dec, dec_rubin)
+
+    x1, y1 = wcs.radecToxy(ra, dec, units=galsim.degrees)
+    assert_allclose(x, x1)
+    assert_allclose(y, y1)
+
+
+def test_rubin_sky_wcs_equal():
+    wcs1 = seacliff.RubinSkyWCS(_make_wcs(10), origin=galsim.PositionD(10, 2))
+    assert wcs1 == wcs1
+    _test_pos(wcs1, wcs1)
+
+    wcs2 = seacliff.RubinSkyWCS(_make_wcs(10), origin=galsim.PositionD(10, 2))
+    assert wcs1 == wcs2
+    assert wcs2 is not wcs1
+    _test_pos(wcs1, wcs2)
+
+    wcs3 = seacliff.RubinSkyWCS(_make_wcs(10))
+    assert wcs3 == wcs3
+    assert wcs1 != wcs3
+    assert wcs2 != wcs3
+    assert wcs3 is not wcs1
+
+    wcs4 = seacliff.RubinSkyWCS(_make_wcs(1))
+    assert wcs1 != wcs4
+    assert wcs2 != wcs4
+    assert wcs3 != wcs4
+    assert wcs4 == wcs4
+    assert wcs4 is not wcs1
+
+
+def test_rubin_sky_wcs_eval_repr():
+    wcs1 = seacliff.RubinSkyWCS(_make_wcs(10), origin=galsim.PositionD(10, 2))
+    wcs5 = eval(repr(wcs1))
+    assert wcs5 == wcs1
+    assert wcs5 is not wcs1
+
+    _test_pos(wcs1, wcs5)
+
+
+def test_rubin_sky_wcs_copy():
+    wcs1 = seacliff.RubinSkyWCS(_make_wcs(10), origin=galsim.PositionD(10, 2))
+    wcs6 = wcs1.copy()
+    assert wcs6 == wcs1
+    assert wcs6 is not wcs1
+    assert wcs1.wcs is not wcs6.wcs
+
+    _test_pos(wcs1, wcs6)
+
+
+def test_rubin_sky_wcs_pickles():
+    wcs1 = seacliff.RubinSkyWCS(_make_wcs(10), origin=galsim.PositionD(10, 2))
+    b = io.BytesIO()
+    pickle.dump(wcs1, b)
+    b.seek(0)
+    wcs2 = pickle.load(b)
```

Comment:
It may be worth porting over our "do_pickle" test function that we have in GalSim, which tests ==, pickle, copy, deepcopy, and eval(repr()) all together.  We run this on all our objects.  Doing this kind of thing separately on every class gets old...
https://github.com/GalSim-developers/GalSim/blob/releases/2.3/tests/galsim_test_helpers.py#L399